File: android/rctmapboxar/src/main/java/com/mapbox/react/ar/geometries/PlaneGeometry.java
package com.mapbox.react.ar.geometries;

import com.mapbox.react.ar.core.Geometry;

public class PlaneGeometry extends Geometry {
    public int width;
    public int height;
    public int widthSegments;
    public int heightSegments;

    public PlaneGeometry(int width, int height, int widthSegments, int heightSegments) {
        super();

        this.width = width;
        this.height = height;
        this.widthSegments = widthSegments;
        this.heightSegments = heightSegments;

        this.fromBufferGeometry(new PlaneBufferGeometry(width, height, widthSegments, heightSegments));
    }
}

File: ios/RNMapboxAR/PlaneGeometry.m
//
//  PlaneGeometry.m
//  RNLocateTest
//
//  Created by Dave Prukop on 4/30/18.
//  Copyright © 2018 Facebook. All rights reserved.
//

#import "PlaneGeometry.h"
#import "PlaneBufferGeometry.h"

@implementation PlaneGeometry

- (instancetype)initWithWidth:(int)width andHeight:(int)height andWidthSegments:(int)widthSegments andHeightSegments:(int)heightSegments
{
  self = [super init];
  if (self) {
    self.width = width;
    self.height = height;
    self.widthSegments = widthSegments;
    self.heightSegments= heightSegments;
    
    [self fromBufferGeometry:[[PlaneBufferGeometry alloc] initWithWidth:width andHeight:height andWidthSegments:widthSegments andHeightSegments:heightSegments]];
  }
  return self;
}

@end
